rest_framework_swagger/tests.py

```diff
@@ -651,6 +651,49 @@ def post(self, request, *args, **kwargs):
         self.assertNotIn('parameters', post)
         self.assertIn('--iron-socks', post['notes'])
 
+    def get_introspector_test(self):
+        """
+        yaml method specifications are dependent on all patterns
+        """
+        class MyViewSet(ModelViewSet):
+            """
+            ---
+            list:
+                param: my param
+            retrieve:
+                param: my param
+            """
+            serializer_class = CommentSerializer
+            model = User
+
+        # Test an object endpoint
+        url_patterns = patterns(
+            '',
+            url(
+                r'^/api/endpoint/(?P<pk>[^/]+)$',
+                MyViewSet.as_view({
+                    'get': 'retrieve',
+                    'put': 'update',
+                    'patch': 'partial_update',
+                    'delete': 'destroy'
+                })
+            ),
+            url(
+                r'^/api/endpoint/$',
+                MyViewSet.as_view({
+                    'get': 'list',
+                    'post': 'create',
+                }))
+        )
+        urlparser = UrlParser()
+        apis = urlparser.get_apis(url_patterns)
+        api = apis[0]
+        self.assertIn('pk', api['pattern'].regex.pattern)
+        generator = DocumentationGenerator()
+        introspector = generator.get_introspector(api, apis)
```

Comment:
I do not validate the test case, because you are supposed to call `generator.get_introspector(api)` 2 times, one for each view.

When I check the current master `docgenerator.DocumentationGenerator.get_introspector` is called by `DocumentationGenerator._get_serializer_set` and `DocumentationGenerator.generate` (through `DocumentationGenerator.get_operations`) and both of them are iterating over `apis`. It means all views will be parsed, thus there is not need to pass over the `apis` argument.

Please can you explain me what problem do you have with the current version of master ? I would like to help